### DISCUSSION TITLE
why etcd serving as a standard go-module project failed to run "./build.sh" in root path

### DISCUSSION THREAD

**Discussion by pikehuang** [etcd-io/etcd#18909] [Q&A] [answered]
### Bug report criteria

- [X] This bug report is not security related, security issues should be disclosed privately via [etcd maintainers](mailto:etcd-maintainers@googlegroups.com).
- [ ] This is not a support request or question, support requests or questions should be raised in the etcd [discussion forums](https://github.com/etcd-io/etcd/discussions).
- [ ] You have read the etcd [bug reporting guidelines](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/reporting_bugs.md).
- [ ] Existing open issues along with etcd [frequently asked questions](https://etcd.io/docs/latest/faq) have been checked and this is not a duplicate.

### What happened?

when run ./build.sh in etcd root directory, it failed and gave error:
must be run from 'go.etcd.io/etcd/v3' module directory

the etcd project is cloned from github and didn't have any modification.

### What did you expect to happen?

./build.sh run success at one shot

### How can we reproduce it (as minimally and precisely as possible)?

step 1:
git clone https://github.com/etcd-io/etcd.git
git checkout release-3.5

step 2:
cd etcd
./build.sh  then errors:
must be run from 'go.etcd.io/etcd/v3' module directory

### Anything else we need to know?

env： x86_64 GNU/Linux

### Etcd version (please run commands below)

<details>

```console
$ etcd --version
# paste output here

$ etcdctl version
# paste output here
```

nothing relevant to etcd and etcdctl, just the start up of building ectcd binary from etcd repo is not fluently.
</details>

### Etcd configuration (command line flags or environment variables)

<details>

# paste your configuration here

</details>

### Etcd debug information (please run commands below, feel free to obfuscate the IP address or FQDN in the output)

<details>

```console
$ etcdctl member list -w table
# paste output here

$ etcdctl --endpoints=<member list> endpoint status -w table
# paste output here
```

</details>

**Comment by ivanvc** [comment]
Hi @pikehuang, thanks for raising this issue. I'll convert this into a discussion so we can gather more information and confirm if a bug is present.

**Comment by ivanvc** [comment]
Hi @pikehuang, can you please let us know what Go version you are using? I cannot reproduce the issue using Go 1.22.9 (`.go-version` file). Thanks.

**Comment by pikehuang** [comment, marked as answer]
hi, ivanvc, from your response I repeated the build process in my personal computer environment instead of my company working computer, then ./build.sh run pass without any problem. So the result is obvious: the working computer access the git repo using golang proxy, which leads to errors as I reported above. 

So the problem can be closed. thanks for your reply.
